File: model/src/main/scala/com/lightbend/model/scala/ModelFactory.scala
package com.lightbend.model.scala

import com.lightbend.modelserver.support.scala.ModelToServe


/**
  * Created by boris on 5/9/17.
  * Basic trait for model factory
  */
trait ModelFactory {
  def create(input : ModelToServe) : Option[Model]
  def restore(bytes : Array[Byte]) : Model
}

File: model/src/main/java/com/lightbend/model/java/ModelFactory.java
package com.lightbend.model.java;

import com.lightbend.modelserver.support.java.ModelToServe;

import java.util.Optional;

/**
 * Created by boris on 7/14/17.
 */
public interface ModelFactory {
    Optional<Model> create(ModelToServe descriptor);
    Model restore(byte[] bytes);
}
